src/main/java/org/terasology/launcher/model/GameIdentifier.java

```diff
@@ -81,7 +85,8 @@ public static GameIdentifier fromString(String identifier) {
             final Build build = Build.valueOf(matcher.group("build"));
             final Profile profile = Profile.valueOf(matcher.group("profile"));
             final String version = matcher.group("version");
-            return new GameIdentifier(version, build, profile);
+            Semver engineVersion = new Semver(version, Semver.SemverType.IVY); // FIXME: assumes engineVersion==version!
```

Comment:
GameIdentifier.fromString has a comment saying

> TODO: move deserialization somewhere else (probably to LauncherSettings?)

It's used by BaseLauncherSettings.getLastPlayedGameVersion, and takes its value from an entry in the settings file (stored in Java properties format).

Reply:
This will break with existing installations of nightly builds :confused: `alpha-20+150` is not a valid semver, and this will just fail...

Reply:
... but on the other hand, re-installing the same game should work, right? :thinking: so, we'd need a way to at least inform the user that they should un-install and install again...

Reply:
nah, it'd break on new installs too, if they're still using `displayVersion` but not `engineVersion` in `GameIdentifier.toString`

Reply:
I changed BaseLauncherSettings to use Gson instead of this fromString method.

I did not include backwards compatibility for the old (non-json) value of this field.

Reply:
This results in an ERROR being logged, even though it does not crash the launcher - would log level WARN be enough? 

A user may see this error multiple times, but on the upside it will be automatically fixed when the user starts a (different?) game :+1: Afterwards, the error logs are gone, so I think not being backwards compatible here is okay.